packages/pyright-internal/src/analyzer/service.ts

```diff
@@ -222,6 +222,22 @@ export class AnalyzerService {
         return service;
     }
 
+    async useEditMode(callback: () => Promise<void>, token: CancellationToken) {
```

Comment:
Why is this async? I think it's important for this to be synchronous.

Reply:
The code in fix all (inside of pylance) is sort of async. It calls common functions that wait for code on the background thread if it exists. Meaning there could be sync versions of those. 

 I agree, we should just enforce synchronous code here. Pylance will just have to change to not make async calls when in edit mode.